internal/provider/tags_data_source.go

```diff
@@ -102,22 +102,41 @@ func (d *TagsDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 		return
 	}
 
-	found := false
+	pkgs := map[string]string{}
+	var found string
 	for _, pkg := range ic.Contents.Packages {
-		if strings.HasPrefix(pkg, data.TargetPackage.ValueString()+"=") {
-			fullVersion := strings.TrimPrefix(pkg, data.TargetPackage.ValueString()+"=")
-			data.Tags = getStemmedVersionTags(fullVersion)
-			data.Tags = append(data.Tags, fullVersion)
-			sort.Strings(data.Tags)
-			found = true
-			break
+		pkg, version, ok := strings.Cut(pkg, "=")
+		if !ok {
+			resp.Diagnostics.AddError("Invalid package", fmt.Sprintf("Invalid package: %s", pkg))
+			return
 		}
+		pkgs[pkg] = version
 	}
-	if !found {
-		resp.Diagnostics.AddError("Unable to find package", fmt.Sprintf("Unable to find package: %s", data.TargetPackage.ValueString()))
+	if _, ok := pkgs[data.TargetPackage.ValueString()]; ok {
+		found = data.TargetPackage.ValueString()
+	} else {
+		for pkg := range pkgs {
+			if strings.HasPrefix(pkg, data.TargetPackage.ValueString()+"-") {
```

Comment:
Could you add a comment here, and in particular give the node-zyx example that this is here to work around that?